dags/financial_data_dag.py

```diff
@@ -0,0 +1,135 @@
+"""
+Downloads stock market data from S&P 500 companies and inserts it into CrateDB.
+
+Prerequisites
+-------------
+In CrateDB, the schema to store this data needs to be created once manually.
+See the file setup/financial_data_schema.sql in this repository.
+
+"""
+import datetime
+import math
+import json
+import requests
+from bs4 import BeautifulSoup
+import yfinance as yf
+import pandas as pd
+
+from airflow import DAG
+from airflow.providers.postgres.operators.postgres import PostgresOperator
+from airflow.operators.python import PythonOperator
+
+def get_sp500_ticker_symbols():
+    """Extracts SP500 companies' tickers from the SP500's wikipedia page"""
+
+    # getting the html code from S&P 500 wikipedia page
+    url = "https://en.wikipedia.org/wiki/List_of_S%26P_500_companies"
+    r_html = requests.get(url,timeout=2.5).text
+    soup = BeautifulSoup(r_html, 'html.parser')
+    components_table = soup.find_all(id="constituents")
+
+    # the first section (index=0) in the components table contains
+    # the headers (which are unimportant in this context) therefore,
+    # only data from index=1 on is taken. moreover, each row in the
+    # table is stored in a <tr> division, so we find all of those.
+    # The data is stored in an array, where each element contains
+    # information about a S&P 500 company.
+    data_rows = components_table[0].find("tbody").find_all("tr")[1:]
+    tickers = []
+
+    # extracting the tickers from the data
+    for row, _ in enumerate(data_rows):
+        stock = list(filter(None,data_rows[row].text.split("\n")))
+        symbol = stock[0]
```

Comment:
I think this part could be simplified a bit. My suggestion would be instead of finding all table rows and then extracting the ticker symbol from the table row to change the beautiful soup behaviour and directly find all first cells of all rows.

This should make it easier to get all stock symbols